components/brave_wallet/browser/json_rpc_service.cc

```diff
@@ -663,17 +663,52 @@ void JsonRpcService::OnFilGetBalance(
   std::move(callback).Run(balance, mojom::ProviderError::kSuccess, "");
 }
 
-void JsonRpcService::GetTransactionCount(const std::string& address,
-                                         GetTxCountCallback callback) {
+void JsonRpcService::GetFilTransactionCount(const std::string& address,
+                                            GetFilTxCountCallback callback) {
   auto internal_callback =
-      base::BindOnce(&JsonRpcService::OnGetTransactionCount,
+      base::BindOnce(&JsonRpcService::OnFilGetTransactionCount,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback));
+  RequestInternal(fil_getTransactionCount(address), true,
+                         network_urls_[mojom::CoinType::ETH],
```

Comment:
FIL

Reply:
Probably a good idea to make sure this function will be using the right network_url so we won't wrongly pass the wrong coin type in the future.

Reply:
done in https://github.com/brave/brave-core/pull/12539/commits/f7ffaf3342953d6313b2f6504fec3bbb722a9dab